File: src/main/java/com/flexpoker/login/repository/LoginRepository.java
package com.flexpoker.login.repository;

import java.util.UUID;

import org.springframework.security.core.userdetails.UserDetailsService;

public interface LoginRepository extends UserDetailsService {

    void saveUsernameAndPassword(String username, String encryptedPassword);

    UUID fetchAggregateIdByUsername(String username);

    void saveAggregateIdAndUsername(UUID aggregateId, String username);

    String fetchUsernameByAggregateId(UUID createdByPlayerId);

}

File: src/main/kotlin/com/flexpoker/login/repository/LoginRepository.kt
package com.flexpoker.login.repository

import org.springframework.security.core.userdetails.UserDetailsService
import java.util.UUID

interface LoginRepository : UserDetailsService {
    fun saveUsernameAndPassword(username: String, encryptedPassword: String)
    fun fetchAggregateIdByUsername(username: String): UUID
    fun saveAggregateIdAndUsername(aggregateId: UUID, username: String)
    fun fetchUsernameByAggregateId(aggregateId: UUID): String
}